model/user.js

```diff
@@ -0,0 +1,27 @@
+module.exports = function(mongoose) {
+  if (!mongoose) {
+    console.error("Mongoose not initialized");
+  }
+  var User = mongoose.model('User', {
+    id: Number,
+    name: String
+  });
+
+  var user = new User({
+    name: 'Bob'
+  });
+  user.save(function(err) {
+    if (err) // ...
+      console.log('user not saved');
+  });
+
+  function findByName(name, callback) {
+    var bob = User.findOne({
+      'name': 'Bob'
+    }, callback);
+  }
+
+  return {
```

Comment:
Mostly basic Mongoose api stuff. This last part is returning a javascript object that is exposing a function (via js closure concept) to the outside world. Everything else is scoped to the main function (i.e. function(mongoose){}) and is harder or impossible to access.